runtime/io.c

```diff
@@ -636,7 +625,7 @@ CAMLprim value caml_ml_out_channels_list (value unit)
     if (channel->max == NULL
         && channel->flags & CHANNEL_FLAG_MANAGED_BY_GC) {
```

Comment:
Shouldn't we remove this part of the test or replace it by an assert now that we only link gc-managed channels?

Reply:
Yes, absolutely.  Will put an assertion.